queries/words.py

```diff
@@ -0,0 +1,238 @@
+    "d": "dé",
+    "ð": "eð",
+    "e": "e",
+    "é": "je",
+    "f": "eff",
+    "g": "gé",
+    "h": "há",
+    "i": "i",
+    "í": "í",
+    "j": "joð",
+    "k": "ká",
+    "l": "ell",
+    "m": "emm",
+    "n": "enn",
+    "o": "o",
+    "ó": "ó",
+    "p": "pé",
+    "q": "kú",
+    "r": "err",
+    "s": "ess",
+    "t": "té",
+    "u": "u",
+    "ú": "ú",
+    "v": "vaff",
+    "x": "ex",
+    "y": "ufsilon i",
+    "ý": "ufsilon í",
+    "þ": "þoddn",
+    "æ": "æ",
+    "ö": "ö",
+    "z": "seta",
+}
+
+
+_SPELLING_RX = (
+    r"^hvernig stafsetur maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig stafsetur maður (.+)$",
+    r"^hvernig skal stafsetja (.+)$",
+    r"^hvernig skal stafsetja (?:orðið|nafnið) (.+)$",
+    r"^hvernig skrifar maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig skrifar maður (.+)$",
+    r"^hvernig stafar maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig stafar maður (.+)$",
+    r"^hvernig er (?:orðið|nafnið) (.+) stafsett$",
+    r"^hvernig er (.+) stafsett$",
+    r"^hvernig er (?:orðið|nafnið) (.+) skrifað$",
+    r"^hvernig er (.+) skrifað$",
+    r"^hvernig er (?:orðið|nafnið) (.+) stafað$",
+    r"^hvernig er (.+) stafað$",
+    r"^hvernig skal stafa (.+)$",
+    r"^hvernig skal stafa (?:orðið|nafnið) (.+)$",
+    r"^hvernig stafast (?:orðið|nafnið) (.+)$",
+)
+
+# Add "nafnið" non-capture groups
+
+_DECLENSION_RX = (
+    r"^hvernig beygi ég (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygirðu (?:orðið|nafnið) (.+)$",
+    r"^hvernig á að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig á ég að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig á maður að beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygir maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygir maður (?:orðið|nafnið) (.+)$",
+    r"^hvernig beygist (?:orðið|nafnið) (.+)$",
+    r"^hvernig skal beygja (?:orðið|nafnið) (.+)$",
+    r"^hvernig er (?:orðið|nafnið) (.+) beygt$",
+    r"^hverjar eru beygingarmyndir (?:orðsins|nafnsins) (.+)$",
+    r"^hvað eru beygingarmyndir (?:orðsins|nafnsins) (.+)$",
+)
+
+
+def lookup_best_word(word):
+    """ Look up word in BÍN, pick right one acc. to a criterion. """
+    res = BIN_Db().lookup_nominative(word)
+    if not res:
+        # Try with uppercase first char
+        res = BIN_Db().lookup_nominative(word.capitalize())
+        if not res:
+            return None
+
+    # OK, we have one or more matching words
+    if len(res) == 1:
+        m = res[0]
+    else:
+        # TODO: Pick best result (prefer nouns vs. adjectives, etc?)
+        m = res[0]  # For now
+
+    # TODO: Fix this. If more than one declesion possible, list them all
+    def sort_by_preference(m_list):
+        """ Discourage rarer declension forms, i.e. ÞGF2 and ÞGF3 """
+        return sorted(m_list, key=lambda m: "2" in m.beyging or "3" in m.beyging)
+
+    # Look up all cases of the word in BÍN
+    with BIN_Db().get_db() as db:
+        nom = m.stofn
+        acc = db.cast_to_accusative(nom, meaning_filter_func=sort_by_preference)
```

Comment:
This could be made more robust, so that the meaning filter func selects only word forms with the same word category and same id as the original nominative one that we're expanding from.